kv/kvserver: TestFlowControlRaftSnapshotV2 failed

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://mesolite.cluster.engflow.com/invocations/default/8903379f-cc50-4774-8597-97f0df21e83e?testReportRun=1&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [b678a18a39e8b1e904f1557ca8f406498fb57820](https://github.com/cockroachdb/cockroach/commits/b678a18a39e8b1e904f1557ca8f406498fb57820):


```
=== RUN   TestFlowControlRaftSnapshotV2
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestFlowControlRaftSnapshotV23122271599
    test_log_scope.go:76: use -show-logs to present logs inline
    flow_control_integration_test.go:924: -- test log scope end --
test logs left over in: outputs.zip/logTestFlowControlRaftSnapshotV23122271599
--- FAIL: TestFlowControlRaftSnapshotV2 (39.55s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1/kvadmission.flow_control.mode=apply_to_elastic
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/raft_snapshot_v1_encoding_apply_to_elastic:8:
         
        output didn't match expected:
        @@ -258,11 +258,10 @@
         
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 0 B                   
           a        | 2        | 0 B                   
        -  a        | 3        | 0 B                   
           a        | 4        | 0 B                   
           a        | 5        | 0 B                   
         
         
         -- Another view of tokens, using /inspectz-backed vtables.
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1/kvadmission.flow_control.mode=apply_to_elastic (10.30s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2 (20.30s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/raft_snapshot_v2_encoding_apply_to_all:1:
         
        output didn't match expected:
        @@ -258,11 +258,10 @@
         
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 0 B                   
           a        | 2        | 0 B                   
        -  a        | 3        | 0 B                   
           a        | 4        | 0 B                   
           a        | 5        | 0 B                   
         
         
         -- Another view of tokens, using /inspectz-backed vtables.
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all (10.26s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1 (19.24s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139229 kv/kvserver: TestFlowControlRaftSnapshotV2 failed [C-bug C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1 target-release-25.1.1 target-release-25.2.0]
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-48326

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19021318?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19021318?buildTab=artifacts#/) on master @ [29f72d48de37f6e2ee46b203f647780f05b9b7c8](https://github.com/cockroachdb/cockroach/commits/29f72d48de37f6e2ee46b203f647780f05b9b7c8):


```
=== RUN   TestFlowControlRaftSnapshotV2
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/3c632028e5b231a582105881dfd5d4b5/logTestFlowControlRaftSnapshotV24284095997
    test_log_scope.go:76: use -show-logs to present logs inline
    flow_control_integration_test.go:924: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/3c632028e5b231a582105881dfd5d4b5/logTestFlowControlRaftSnapshotV24284095997
--- FAIL: TestFlowControlRaftSnapshotV2 (39.28s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2 (18.42s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_elastic
    datadriven.go:144: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/30635/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/raft_snapshot_v2_encoding_apply_to_elastic:1:
         
        output didn't match expected:
        @@ -258,11 +258,10 @@
         
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 0 B                   
           a        | 2        | 0 B                   
        -  a        | 3        | 0 B                   
           a        | 4        | 0 B                   
           a        | 5        | 0 B                   
         
         
         -- Another view of tokens, using /inspectz-backed vtables.
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_elastic (9.62s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139229 kv/kvserver: TestFlowControlRaftSnapshotV2 failed [C-bug C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1 target-release-25.1.1 target-release-25.2.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19022675?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19022675?buildTab=artifacts#/) on master @ [29f72d48de37f6e2ee46b203f647780f05b9b7c8](https://github.com/cockroachdb/cockroach/commits/29f72d48de37f6e2ee46b203f647780f05b9b7c8):


```
=== RUN   TestFlowControlRaftSnapshotV2
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/3c632028e5b231a582105881dfd5d4b5/logTestFlowControlRaftSnapshotV2664873217
    test_log_scope.go:76: use -show-logs to present logs inline
    flow_control_integration_test.go:924: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/3c632028e5b231a582105881dfd5d4b5/logTestFlowControlRaftSnapshotV2664873217
--- FAIL: TestFlowControlRaftSnapshotV2 (38.34s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2 (18.15s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_elastic
    datadriven.go:144: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/raft_snapshot_v2_encoding_apply_to_elastic:1:
         
        output didn't match expected:
        @@ -258,11 +258,10 @@
         
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 0 B                   
           a        | 2        | 0 B                   
        -  a        | 3        | 0 B                   
           a        | 4        | 0 B                   
           a        | 5        | 0 B                   
         
         
         -- Another view of tokens, using /inspectz-backed vtables.
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_elastic (9.23s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139229 kv/kvserver: TestFlowControlRaftSnapshotV2 failed [C-bug C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1 target-release-25.1.1 target-release-25.2.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19026148?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19026148?buildTab=artifacts#/) on master @ [53fc6113e33541346b9e37b8ef0b7a0a41aa1877](https://github.com/cockroachdb/cockroach/commits/53fc6113e33541346b9e37b8ef0b7a0a41aa1877):


```
=== RUN   TestFlowControlRaftSnapshotV2
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/3c632028e5b231a582105881dfd5d4b5/logTestFlowControlRaftSnapshotV2979371960
    test_log_scope.go:76: use -show-logs to present logs inline
    flow_control_integration_test.go:925: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/3c632028e5b231a582105881dfd5d4b5/logTestFlowControlRaftSnapshotV2979371960
--- FAIL: TestFlowControlRaftSnapshotV2 (35.34s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1/kvadmission.flow_control.mode=apply_to_elastic
    datadriven.go:144: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/15478/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/raft_snapshot_v1_encoding_apply_to_elastic:8:
         
        output didn't match expected:
        @@ -258,11 +258,10 @@
         
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 0 B                   
           a        | 2        | 0 B                   
        -  a        | 3        | 0 B                   
           a        | 4        | 0 B                   
           a        | 5        | 0 B                   
         
         
         -- Another view of tokens, using /inspectz-backed vtables.
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1/kvadmission.flow_control.mode=apply_to_elastic (10.23s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1 (18.61s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139229 kv/kvserver: TestFlowControlRaftSnapshotV2 failed [C-bug C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1 target-release-25.1.1 target-release-25.2.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/ec1efd32-72a4-4c09-8bbe-360c402138ad?testReportRun=1&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [a05a56ec16db34644361b2328d889179c3e410c3](https://github.com/cockroachdb/cockroach/commits/a05a56ec16db34644361b2328d889179c3e410c3):


```
=== RUN   TestFlowControlRaftSnapshotV2
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestFlowControlRaftSnapshotV2903520615
    test_log_scope.go:76: use -show-logs to present logs inline
    flow_control_integration_test.go:925: -- test log scope end --
test logs left over in: outputs.zip/logTestFlowControlRaftSnapshotV2903520615
--- FAIL: TestFlowControlRaftSnapshotV2 (40.94s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2 (20.10s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_elastic
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/raft_snapshot_v2_encoding_apply_to_elastic:1:
         
        output didn't match expected:
        @@ -258,11 +258,10 @@
         
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 0 B                   
           a        | 2        | 0 B                   
        -  a        | 3        | 0 B                   
           a        | 4        | 0 B                   
           a        | 5        | 0 B                   
         
         
         -- Another view of tokens, using /inspectz-backed vtables.
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_elastic (10.64s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139229 kv/kvserver: TestFlowControlRaftSnapshotV2 failed [C-bug C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1 target-release-25.1.1 target-release-25.2.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/c659b442-b9e9-46f8-b328-3d66fdbcf136?testReportRun=1&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [1eb87cd23a305418521623cc8059546c92e5c1be](https://github.com/cockroachdb/cockroach/commits/1eb87cd23a305418521623cc8059546c92e5c1be):


```
         -- Observe the total tracked tokens per-stream on n1. There's nothing tracked
         -- for n2 and n3 anymore.
        @@ -201,11 +201,11 @@
         -----------+----------+-----------------------
           a        | 1        | 2.0 MiB               
           a        | 2        | 0 B                   
           a        | 3        | 0 B                   
           a        | 4        | 2.0 MiB               
        -  a        | 5        | 2.0 MiB               
        +  a        | 5        | 1.0 MiB               
         
         
         -- (Allow below-raft admission to proceed.)
         
         
        @@ -222,29 +222,29 @@
           name ASC;
         
           kvflowcontrol.tokens.eval.elastic.available                       | 40 MiB   
           kvflowcontrol.tokens.eval.elastic.deducted                        | 10 MiB   
           kvflowcontrol.tokens.eval.elastic.returned                        | 10 MiB   
        -  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 4.0 MiB  
        +  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 5.0 MiB  
           kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B      
           kvflowcontrol.tokens.eval.regular.available                       | 80 MiB   
           kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB   
           kvflowcontrol.tokens.eval.regular.returned                        | 10 MiB   
        -  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 4.0 MiB  
        +  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 5.0 MiB  
           kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B      
           kvflowcontrol.tokens.send.elastic.available                       | 40 MiB   
           kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB   
           kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B      
           kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
           kvflowcontrol.tokens.send.elastic.returned                        | 10 MiB   
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 4.0 MiB  
        +  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 5.0 MiB  
           kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
           kvflowcontrol.tokens.send.regular.available                       | 80 MiB   
           kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB   
           kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
           kvflowcontrol.tokens.send.regular.returned                        | 10 MiB   
        -  kvflowcontrol.tokens.send.regular.returned.disconnect             | 4.0 MiB  
        +  kvflowcontrol.tokens.send.regular.returned.disconnect             | 5.0 MiB  
           kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B      
         
         
         -- Observe the total tracked tokens per-stream on n1; there should be nothing.
         SELECT
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_all (27.39s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139229 kv/kvserver: TestFlowControlRaftSnapshotV2 failed [C-bug C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1 target-release-25.1.1 target-release-25.2.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19032915?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19032915?buildTab=artifacts#/) on master @ [a6ce420380cf3820f283fe06c4014d8c4dd1eccd](https://github.com/cockroachdb/cockroach/commits/a6ce420380cf3820f283fe06c4014d8c4dd1eccd):


```
=== RUN   TestFlowControlRaftSnapshotV2
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/93fbc720f61b7e85f714b9750f754229/logTestFlowControlRaftSnapshotV24114509486
    test_log_scope.go:76: use -show-logs to present logs inline
    flow_control_integration_test.go:925: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/93fbc720f61b7e85f714b9750f754229/logTestFlowControlRaftSnapshotV24114509486
--- FAIL: TestFlowControlRaftSnapshotV2 (41.26s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1 (20.07s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1/kvadmission.flow_control.mode=apply_to_all
    datadriven.go:144: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/29216/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/raft_snapshot_v1_encoding_apply_to_all:8:
         
        output didn't match expected:
        @@ -258,11 +258,10 @@
         
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 0 B                   
           a        | 2        | 0 B                   
        -  a        | 3        | 0 B                   
           a        | 4        | 0 B                   
           a        | 5        | 0 B                   
         
         
         -- Another view of tokens, using /inspectz-backed vtables.
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1/kvadmission.flow_control.mode=apply_to_all (10.28s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139229 kv/kvserver: TestFlowControlRaftSnapshotV2 failed [C-bug C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1 target-release-25.1.1 target-release-25.2.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19034343?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19034343?buildTab=artifacts#/) on master @ [ce7656c3e9f0b402ed110ee15e27e22f801dec62](https://github.com/cockroachdb/cockroach/commits/ce7656c3e9f0b402ed110ee15e27e22f801dec62):


```
=== RUN   TestFlowControlRaftSnapshotV2
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/93fbc720f61b7e85f714b9750f754229/logTestFlowControlRaftSnapshotV22116854692
    test_log_scope.go:76: use -show-logs to present logs inline
    flow_control_integration_test.go:925: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/93fbc720f61b7e85f714b9750f754229/logTestFlowControlRaftSnapshotV22116854692
--- FAIL: TestFlowControlRaftSnapshotV2 (29.77s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_elastic
    datadriven.go:144: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/29922/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/raft_snapshot_v2_encoding_apply_to_elastic:1:
         
        output didn't match expected:
        @@ -258,11 +258,10 @@
         
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 0 B                   
           a        | 2        | 0 B                   
        -  a        | 3        | 0 B                   
           a        | 4        | 0 B                   
           a        | 5        | 0 B                   
         
         
         -- Another view of tokens, using /inspectz-backed vtables.
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_elastic (7.30s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2 (15.71s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139229 kv/kvserver: TestFlowControlRaftSnapshotV2 failed [C-bug C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1 target-release-25.1.1 target-release-25.2.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19036350?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19036350?buildTab=artifacts#/) on master @ [677a8a68c85167cdd606e50d72d8522c35bc155c](https://github.com/cockroachdb/cockroach/commits/677a8a68c85167cdd606e50d72d8522c35bc155c):


```
=== RUN   TestFlowControlRaftSnapshotV2
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/93fbc720f61b7e85f714b9750f754229/logTestFlowControlRaftSnapshotV24132234333
    test_log_scope.go:76: use -show-logs to present logs inline
    flow_control_integration_test.go:925: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/93fbc720f61b7e85f714b9750f754229/logTestFlowControlRaftSnapshotV24132234333
--- FAIL: TestFlowControlRaftSnapshotV2 (38.85s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1/kvadmission.flow_control.mode=apply_to_elastic
    datadriven.go:144: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/30022/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/raft_snapshot_v1_encoding_apply_to_elastic:8:
         
        output didn't match expected:
        @@ -258,11 +258,10 @@
         
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 0 B                   
           a        | 2        | 0 B                   
        -  a        | 3        | 0 B                   
           a        | 4        | 0 B                   
           a        | 5        | 0 B                   
         
         
         -- Another view of tokens, using /inspectz-backed vtables.
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1/kvadmission.flow_control.mode=apply_to_elastic (10.25s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1 (18.55s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139229 kv/kvserver: TestFlowControlRaftSnapshotV2 failed [C-bug C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1 target-release-25.1.1 target-release-25.2.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/1e9190f3-1b92-4846-bc41-b5e1c9970ccc?testReportRun=3&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [bd6289d21debf18c3bc0f2889bfe98646c60d1d0](https://github.com/cockroachdb/cockroach/commits/bd6289d21debf18c3bc0f2889bfe98646c60d1d0):


```
=== RUN   TestFlowControlRaftSnapshotV2
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestFlowControlRaftSnapshotV23906742242
    test_log_scope.go:76: use -show-logs to present logs inline
    flow_control_integration_test.go:924: -- test log scope end --
test logs left over in: outputs.zip/logTestFlowControlRaftSnapshotV23906742242
--- FAIL: TestFlowControlRaftSnapshotV2 (40.59s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1 (21.56s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1/kvadmission.flow_control.mode=apply_to_all
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/raft_snapshot_v1_encoding_apply_to_all:8:
         
        output didn't match expected:
        @@ -258,11 +258,10 @@
         
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 0 B                   
           a        | 2        | 0 B                   
        -  a        | 3        | 0 B                   
           a        | 4        | 0 B                   
           a        | 5        | 0 B                   
         
         
         -- Another view of tokens, using /inspectz-backed vtables.
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1/kvadmission.flow_control.mode=apply_to_all (10.70s)
=== RUN   TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1/kvadmission.flow_control.mode=apply_to_elastic
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/raft_snapshot_v1_encoding_apply_to_elastic:8:
         
        output didn't match expected:
        @@ -258,11 +258,10 @@
         
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 0 B                   
           a        | 2        | 0 B                   
        -  a        | 3        | 0 B                   
           a        | 4        | 0 B                   
           a        | 5        | 0 B                   
         
         
         -- Another view of tokens, using /inspectz-backed vtables.
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=1/kvadmission.flow_control.mode=apply_to_elastic (10.86s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139229 kv/kvserver: TestFlowControlRaftSnapshotV2 failed [C-bug C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1 target-release-25.1.1 target-release-25.2.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/1599ed20-5107-4216-8296-97806ec0c222?testReportRun=1&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [bd6289d21debf18c3bc0f2889bfe98646c60d1d0](https://github.com/cockroachdb/cockroach/commits/bd6289d21debf18c3bc0f2889bfe98646c60d1d0):


```
           kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
        -  kvflowcontrol.tokens.send.elastic.returned                        | 4.0 MiB  
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 4.0 MiB  
        +  kvflowcontrol.tokens.send.elastic.returned                        | 5.0 MiB  
        +  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 5.0 MiB  
           kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
           kvflowcontrol.tokens.send.regular.available                       | 80 MiB   
           kvflowcontrol.tokens.send.regular.deducted                        | 0 B      
           kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
           kvflowcontrol.tokens.send.regular.returned                        | 0 B      
        @@ -200,11 +200,11 @@
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 2.0 MiB               
           a        | 2        | 0 B                   
           a        | 3        | 0 B                   
        -  a        | 4        | 2.0 MiB               
        +  a        | 4        | 1.0 MiB               
           a        | 5        | 2.0 MiB               
         
         
         -- (Allow below-raft admission to proceed.)
         
        @@ -222,11 +222,11 @@
           name ASC;
         
           kvflowcontrol.tokens.eval.elastic.available                       | 40 MiB   
           kvflowcontrol.tokens.eval.elastic.deducted                        | 10 MiB   
           kvflowcontrol.tokens.eval.elastic.returned                        | 10 MiB   
        -  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 4.0 MiB  
        +  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 5.0 MiB  
           kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B      
           kvflowcontrol.tokens.eval.regular.available                       | 80 MiB   
           kvflowcontrol.tokens.eval.regular.deducted                        | 0 B      
           kvflowcontrol.tokens.eval.regular.returned                        | 0 B      
           kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B      
        @@ -234,11 +234,11 @@
           kvflowcontrol.tokens.send.elastic.available                       | 40 MiB   
           kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB   
           kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B      
           kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
           kvflowcontrol.tokens.send.elastic.returned                        | 10 MiB   
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 4.0 MiB  
        +  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 5.0 MiB  
           kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
           kvflowcontrol.tokens.send.regular.available                       | 80 MiB   
           kvflowcontrol.tokens.send.regular.deducted                        | 0 B      
           kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
           kvflowcontrol.tokens.send.regular.returned                        | 0 B      
--- FAIL: TestFlowControlRaftSnapshotV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_elastic (27.08s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139229 kv/kvserver: TestFlowControlRaftSnapshotV2 failed [C-bug C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1 target-release-25.1.1 target-release-25.2.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/5b280b59-5726-4b73-a2bf-f077c0e07958?testReportRun=1&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [2362ac69262c53be08f010fcde76371592d311ef](https://github.com/cockroachdb/cockroach/commits/2362ac69262c53be08f010fcde76371592d311ef):


```
        -ORDER BY
        -  name ASC;
        -
        -  kvflowcontrol.tokens.eval.elastic.available                       | 30 MiB  
        -  kvflowcontrol.tokens.eval.elastic.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.eval.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.eval.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.eval.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.eval.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.available                       | 30 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B     
        -  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.regular.available                       | 70 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted                        | 10 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B     
        -
        -
        --- Observe the total tracked tokens per-stream on n1. 2MiB is tracked for n1-n5;
        --- see last comment for an explanation why we're still deducting for n2, n3.
        -SELECT
        -  chr(96 + dense_rank() OVER (ORDER BY range_id)) as range_id,
        -  store_id,
        -  crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
        -FROM
        -  crdb_internal.kv_flow_control_handles_v2
        -ORDER BY
        -  range_id, store_id;
        -
        -  range_id | store_id | total_tracked_tokens  
        ------------+----------+-----------------------
        -  a        | 1        | 2.0 MiB               
        -  a        | 2        | 2.0 MiB               
        -  a        | 3        | 2.0 MiB               
           a        | 4        | 2.0 MiB               
           a        | 5        | 2.0 MiB               
         
         
         -- (Truncating raft log.)
--- FAIL: TestFlowControlRaftSnapshotV2/kvadmission.flow_control.mode=apply_to_all (26.86s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139229 kv/kvserver: TestFlowControlRaftSnapshotV2 failed [C-bug C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1 target-release-25.1.1 target-release-25.2.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/b140e9be-8d83-4eac-9470-571083ddc264?testReportRun=3&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [104eff4b76905adeb7de221fbce8a8b2bca7460f](https://github.com/cockroachdb/cockroach/commits/104eff4b76905adeb7de221fbce8a8b2bca7460f):


```
=== RUN   TestFlowControlRaftSnapshotV2
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestFlowControlRaftSnapshotV2598295885
    test_log_scope.go:76: use -show-logs to present logs inline
    flow_control_integration_test.go:880: -- test log scope end --
test logs left over in: outputs.zip/logTestFlowControlRaftSnapshotV2598295885
--- FAIL: TestFlowControlRaftSnapshotV2 (24.06s)
=== RUN   TestFlowControlRaftSnapshotV2/kvadmission.flow_control.mode=apply_to_elastic
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/raft_snapshot_v2_encoding_apply_to_elastic:1:
         
        output didn't match expected:
        @@ -258,11 +258,10 @@
         
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 0 B                   
           a        | 2        | 0 B                   
        -  a        | 3        | 0 B                   
           a        | 4        | 0 B                   
           a        | 5        | 0 B                   
         
         
         -- Another view of tokens, using /inspectz-backed vtables.
--- FAIL: TestFlowControlRaftSnapshotV2/kvadmission.flow_control.mode=apply_to_elastic (12.70s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139229 kv/kvserver: TestFlowControlRaftSnapshotV2 failed [C-bug C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1 target-release-25.1.1 target-release-25.2.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestFlowControlRaftSnapshotV2 [failed](https://tanzanite.cluster.engflow.com/invocations/default/81119ac1-cac8-4a7b-9d9a-bb9f4b8de7c6?testReportRun=1&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [104eff4b76905adeb7de221fbce8a8b2bca7460f](https://github.com/cockroachdb/cockroach/commits/104eff4b76905adeb7de221fbce8a8b2bca7460f):


```
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B     
        -  kvflowcontrol.tokens.send.regular.available                       | 80 MiB  
        -  kvflowcontrol.tokens.send.regular.deducted                        | 0 B     
        -  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned                        | 0 B     
        -  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B     
        -  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B     
        +  kvflowcontrol.tokens.eval.elastic.available                       | 31 MiB   
        +  kvflowcontrol.tokens.eval.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.eval.elastic.returned                        | 1.0 MiB  
        +  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 1.0 MiB  
        +  kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.eval.regular.available                       | 80 MiB   
        +  kvflowcontrol.tokens.eval.regular.deducted                        | 0 B      
        +  kvflowcontrol.tokens.eval.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.eval.regular.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.available                       | 31 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted                        | 10 MiB   
        +  kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B      
        +  kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.elastic.returned                        | 1.0 MiB  
        +  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 1.0 MiB  
        +  kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B      
        +  kvflowcontrol.tokens.send.regular.available                       | 80 MiB   
        +  kvflowcontrol.tokens.send.regular.deducted                        | 0 B      
        +  kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned                        | 0 B      
        +  kvflowcontrol.tokens.send.regular.returned.disconnect             | 0 B      
        +  kvflowcontrol.tokens.send.regular.unaccounted                     | 0 B      
         
         
         -- Observe the total tracked tokens per-stream on n1. 2MiB is tracked for n1-n5;
         -- see last comment for an explanation why we're still deducting for n2, n3.
         SELECT
        @@ -133,11 +133,11 @@
           range_id, store_id;
         
           range_id | store_id | total_tracked_tokens  
         -----------+----------+-----------------------
           a        | 1        | 2.0 MiB               
        -  a        | 2        | 2.0 MiB               
        +  a        | 2        | 1.0 MiB               
           a        | 3        | 2.0 MiB               
           a        | 4        | 2.0 MiB               
           a        | 5        | 2.0 MiB               
         
         
--- FAIL: TestFlowControlRaftSnapshotV2/kvadmission.flow_control.mode=apply_to_elastic (27.49s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139229 kv/kvserver: TestFlowControlRaftSnapshotV2 failed [C-bug C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1 target-release-25.1.1 target-release-25.2.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlRaftSnapshotV2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/kvserver/flow_control_integration_test.go
+++ b/pkg/kv/kvserver/flow_control_integration_test.go
@@ -79,6 +79,9 @@ func TestFlowControlBasicV2(t *testing.T) {
 					Settings: settings,
 					Knobs: base.TestingKnobs{
 						Store: &kvserver.StoreTestingKnobs{
+							RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+								return true
+							},
 							FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 								UseOnlyForScratchRanges: true,
 								OverrideTokenDeduction: func(_ kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -172,6 +175,9 @@ func TestFlowControlRangeSplitMergeV2(t *testing.T) {
 				Settings: settings,
 				Knobs: base.TestingKnobs{
 					Store: &kvserver.StoreTestingKnobs{
+						RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+							return true
+						},
 						FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 							UseOnlyForScratchRanges: true,
 							OverrideTokenDeduction: func(_ kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -291,6 +297,9 @@ func TestFlowControlBlockedAdmissionV2(t *testing.T) {
 				Settings: settings,
 				Knobs: base.TestingKnobs{
 					Store: &kvserver.StoreTestingKnobs{
+						RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+							return true
+						},
 						FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 							UseOnlyForScratchRanges: true,
 							OverrideTokenDeduction: func(_ kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -401,6 +410,9 @@ func TestFlowControlAdmissionPostSplitMergeV2(t *testing.T) {
 						WallClock: manualClock,
 					},
 					Store: &kvserver.StoreTestingKnobs{
+						RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+							return true
+						},
 						FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 							UseOnlyForScratchRanges: true,
 							OverrideTokenDeduction: func(_ kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -536,6 +548,11 @@ func TestFlowControlCrashedNodeV2(t *testing.T) {
 		// virtue of raft fortification) help ensure this. Override to disable any
 		// metamorphosis.
 		kvserver.OverrideDefaultLeaseType(ctx, &settings.SV, roachpb.LeaseLeader)
+
+		// We want to observe a disconnected stream when crashing a node, however
+		// we don't care about replicas blipping into StateProbe otherwise.
+		var bypassReplicaUnreachable atomic.Bool
+		bypassReplicaUnreachable.Store(true)
 		tc := testcluster.StartTestCluster(t, 3, base.TestClusterArgs{
 			ReplicationMode: base.ReplicationManual,
 			ServerArgs: base.TestServerArgs{
@@ -547,6 +564,9 @@ func TestFlowControlCrashedNodeV2(t *testing.T) {
 				},
 				Knobs: base.TestingKnobs{
 					Store: &kvserver.StoreTestingKnobs{
+						RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+							return bypassReplicaUnreachable.Load()
+						},
 						FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 							UseOnlyForScratchRanges: true,
 							OverrideTokenDeduction: func(_ kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -603,9 +623,11 @@ func TestFlowControlCrashedNodeV2(t *testing.T) {
 		h.comment(`-- Observe the per-stream tracked tokens on n1, before n2 is crashed.`)
 		h.query(n1, v2FlowPerStreamTrackedQueryStr, flowPerStreamTrackedQueryHeaderStrs...)
 
+		bypassReplicaUnreachable.Store(false)
 		h.comment(`-- (Crashing n2)`)
 		tc.StopServer(1)
 		h.waitForConnectedStreams(ctx, desc.RangeID, 2, 0 /* serverIdx */)
+		bypassReplicaUnreachable.Store(true)
 
 		h.comment(`
 -- Observe the per-stream tracked tokens on n1, after n2 crashed. We're no
@@ -638,8 +660,6 @@ func TestFlowControlRaftSnapshotV2(t *testing.T) {
 		stickyServerArgs := make(map[int]base.TestServerArgs)
 		var disableWorkQueueGranting atomic.Bool
 		disableWorkQueueGranting.Store(true)
-		var bypassReplicaUnreachable atomic.Bool
-		bypassReplicaUnreachable.Store(false)
 		ctx := context.Background()
 		settings := cluster.MakeTestingClusterSettings()
 		// This test doesn't want leadership changing hands, and leader leases (by
@@ -666,12 +686,7 @@ func TestFlowControlRaftSnapshotV2(t *testing.T) {
 					},
 					Store: &kvserver.StoreTestingKnobs{
 						RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
-							// This test is going to crash nodes, then truncate the raft log
-							// and assert that tokens are returned upon an replica entering
-							// StateSnapshot. To avoid the stopped replicas entering
-							// StateProbe returning tokens, we disable reporting a replica
-							// as unreachable while nodes are down.
-							return bypassReplicaUnreachable.Load()
+							return true
 						},
 						FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 							UseOnlyForScratchRanges: true,
@@ -776,12 +791,6 @@ func TestFlowControlRaftSnapshotV2(t *testing.T) {
 		// Kill stores 1 + 2, increment the key on the other stores and truncate
 		// their logs to make sure that when store 1 + 2 comes back up they will
 		// require a snapshot from Raft.
-		//
-		// Also prevent replicas on the killed nodes from being marked as
-		// unreachable, in order to prevent them from returning tokens via
-		// entering StateProbe, before we're able to truncate the log and assert
-		// on the snapshot behavior.
-		bypassReplicaUnreachable.Store(true)
 		tc.StopServer(1)
 		tc.StopServer(2)
 
@@ -833,7 +842,6 @@ func TestFlowControlRaftSnapshotV2(t *testing.T) {
 		h.comment(`-- (Restarting n2 and n3.)`)
 		require.NoError(t, tc.RestartServer(1))
 		require.NoError(t, tc.RestartServer(2))
-		bypassReplicaUnreachable.Store(false)
 
 		tc.WaitForValues(t, k, []int64{incAB, incAB, incAB, incAB, incAB})
 
@@ -907,6 +915,9 @@ func TestFlowControlRaftMembershipV2(t *testing.T) {
 				Settings: settings,
 				Knobs: base.TestingKnobs{
 					Store: &kvserver.StoreTestingKnobs{
+						RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+							return true
+						},
 						FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 							UseOnlyForScratchRanges: true,
 							OverrideTokenDeduction: func(_ kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -1036,6 +1047,9 @@ func TestFlowControlRaftMembershipRemoveSelfV2(t *testing.T) {
 					Settings: settings,
 					Knobs: base.TestingKnobs{
 						Store: &kvserver.StoreTestingKnobs{
+							RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+								return true
+							},
 							FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 								UseOnlyForScratchRanges: true,
 								OverrideTokenDeduction: func(_ kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -1165,6 +1179,9 @@ func TestFlowControlClassPrioritizationV2(t *testing.T) {
 				Settings: settings,
 				Knobs: base.TestingKnobs{
 					Store: &kvserver.StoreTestingKnobs{
+						RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+							return true
+						},
 						FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 							UseOnlyForScratchRanges: true,
 							OverrideTokenDeduction: func(_ kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -1268,6 +1285,9 @@ func TestFlowControlUnquiescedRangeV2(t *testing.T) {
 				},
 				Knobs: base.TestingKnobs{
 					Store: &kvserver.StoreTestingKnobs{
+						RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+							return true
+						},
 						FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 							UseOnlyForScratchRanges: true,
 							OverrideTokenDeduction: func(_ kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -1382,6 +1402,9 @@ func TestFlowControlTransferLeaseV2(t *testing.T) {
 				Settings: settings,
 				Knobs: base.TestingKnobs{
 					Store: &kvserver.StoreTestingKnobs{
+						RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+							return true
+						},
 						FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 							UseOnlyForScratchRanges: true,
 							OverrideTokenDeduction: func(_ kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -1469,6 +1492,9 @@ func TestFlowControlLeaderNotLeaseholderV2(t *testing.T) {
 				Settings: settings,
 				Knobs: base.TestingKnobs{
 					Store: &kvserver.StoreTestingKnobs{
+						RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+							return true
+						},
 						// Disable leader transfers during leaseholder changes so
 						// that we can easily create leader-not-leaseholder
 						// scenarios.
@@ -1584,6 +1610,9 @@ func TestFlowControlGranterAdmitOneByOneV2(t *testing.T) {
 				Settings: settings,
 				Knobs: base.TestingKnobs{
 					Store: &kvserver.StoreTestingKnobs{
+						RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+							return true
+						},
 						FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 							UseOnlyForScratchRanges: true,
 							OverrideTokenDeduction: func(_ kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -1723,6 +1752,9 @@ func TestFlowControlSendQueue(t *testing.T) {
 					WallClock:         manualClock,
 				},
 				Store: &kvserver.StoreTestingKnobs{
+					RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+						return true
+					},
 					FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 						UseOnlyForScratchRanges: true,
 						OverrideTokenDeduction: func(tokens kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -2056,6 +2088,9 @@ func TestFlowControlRepeatedlySwitchMode(t *testing.T) {
 			Settings: settings,
 			Knobs: base.TestingKnobs{
 				Store: &kvserver.StoreTestingKnobs{
+					RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+						return true
+					},
 					FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 						UseOnlyForScratchRanges: true,
 						OverrideTokenDeduction: func(_ kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -2141,6 +2176,9 @@ func TestFlowControlSendQueueManyInflight(t *testing.T) {
 			Settings: settings,
 			Knobs: base.TestingKnobs{
 				Store: &kvserver.StoreTestingKnobs{
+					RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+						return true
+					},
 					FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 						UseOnlyForScratchRanges: true,
 						OverrideTokenDeduction: func(tokens kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -2334,6 +2372,9 @@ func TestFlowControlSendQueueRangeRelocate(t *testing.T) {
 					Settings: settings,
 					Knobs: base.TestingKnobs{
 						Store: &kvserver.StoreTestingKnobs{
+							RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+								return true
+							},
 							FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 								UseOnlyForScratchRanges: true,
 								OverrideTokenDeduction: func(tokens kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -2554,6 +2595,9 @@ func TestFlowControlRangeSplitMergeMixedVersion(t *testing.T) {
 					DisableAutomaticVersionUpgrade: make(chan struct{}),
 				},
 				Store: &kvserver.StoreTestingKnobs{
+					RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+						return true
+					},
 					FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 						UseOnlyForScratchRanges: true,
 						OverrideTokenDeduction: func(tokens kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -2796,6 +2840,9 @@ func TestFlowControlSendQueueRangeMigrate(t *testing.T) {
 					DisableAutomaticVersionUpgrade: make(chan struct{}),
 				},
 				Store: &kvserver.StoreTestingKnobs{
+					RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+						return true
+					},
 					EvalKnobs: kvserverbase.BatchEvalTestingKnobs{
 						// Because we are migrating from a version (currently) prior to the
 						// range force flush key version gate, we won't trigger the force
@@ -3008,6 +3055,9 @@ func TestFlowControlSendQueueRangeSplitMerge(t *testing.T) {
 			Settings: settings,
 			Knobs: base.TestingKnobs{
 				Store: &kvserver.StoreTestingKnobs{
+					RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+						return true
+					},
 					FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 						UseOnlyForScratchRanges: true,
 						OverrideTokenDeduction: func(tokens kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -3253,6 +3303,9 @@ func TestFlowControlSendQueueRangeFeed(t *testing.T) {
 			Settings: settings,
 			Knobs: base.TestingKnobs{
 				Store: &kvserver.StoreTestingKnobs{
+					RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+						return true
+					},
 					FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 						UseOnlyForScratchRanges: true,
 						OverrideTokenDeduction: func(tokens kvflowcontrol.Tokens) kvflowcontrol.Tokens {
@@ -3964,6 +4017,9 @@ func BenchmarkFlowControlV2Basic(b *testing.B) {
 				Settings: settings,
 				Knobs: base.TestingKnobs{
 					Store: &kvserver.StoreTestingKnobs{
+						RaftReportUnreachableBypass: func(_ roachpb.ReplicaID) bool {
+							return true
+						},
 						FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
 							UseOnlyForScratchRanges: true,
 							OverrideTokenDeduction: func(_ kvflowcontrol.Tokens) kvflowcontrol.Tokens {

